packages/core/src/controllers/ModalController.ts

```diff
@@ -67,9 +68,21 @@ export const ModalController = {
   },
 
   close() {
+    const isEmbeddedEnabled = OptionsController.state.experimental_enableEmbedded
     const connected = Boolean(ChainController.state.activeCaipAddress)
+
     state.open = false
-    PublicStateController.set({ open: false })
+
+    if (isEmbeddedEnabled) {
```

Comment:
This is annoying but I didn't want to make things fancier. For specifically Builder app, we are rendring the modal inside the page, rather than a modal. For embedded mode, we should redirect user to relevant pages depending on the connection states, instead of closing the modal.

I've added `experimental_enableEmbedded` flag to control this.